### PR TITLE
Feat/common module

### PR DESCRIPTION
**Pull request by gluax**
## Motivation

To have commonly used Data Structures from a single source rather than re-defining them.

## Explanation of Changes

This creates a new module within our workspace, `seda-common`, and we define the commonly used structures there.
This module exposes a `cli` flag that provides clap bindings.

The `contracts` and `cli` modules now use these as a single source of truth. 

The update command has been changed to be a struct enum in a single place so that it also works for the CLI.

It also adds a `test` feature to the module so that we can generate better mock data for our mock rpc.

**NOTES**: 

- ~~The register command needs the private key and signature. I have dummy values there since we don't store the `KeyPair` on the  `NodeConfig` in this branch. Once that change is in, we can address that.~~
- Similarly, the `unregister` command has been removed from the contract but is to be added back in. So, for now, I'm not making any changes to it, and when we add that command back in, we can fix it.

## Testing

By hand.

## Related PRs and Issues

~~#85 adds the `KeyPair` in the `NodeConfig`, so that PR can make the changes to fix the `register` command if this one is merged first. Or we can do it in a separate follow-up PR. ~~
~~#91 Has changes made to some of these structs in its PR, so there will be conflicts one way or another after one is merged.~~

### CODE REVIEW

File: seda_sk
Comment (jamesondh) [inline seda_sk:1]: should be .gitignored?

**Review by jamesondh — CHANGES_REQUESTED** [review]

File: seda_sk
Comment (FranklinWaller) [reply seda_sk:1]: Did a secret key just got leaked 😛

File: common/src/node/get.rs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+impl ToString for GetNodeArgs {
+    fn to_string(&self) -> String {
+        let json = json!(self);
+        json.to_string()
+    }
+}
+
+#[derive(Debug, Clone, Default, Eq, PartialEq, BorshDeserialize, BorshSerialize, Deserialize, Serialize)]
+
```
Comment (mariocao) [inline common/src/node/get.rs:22]:
```suggestion
```

File: common/src/node/mod.rs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+mod register;
+pub use register::*;
+
+mod update;
+pub use update::*;
+
+use super::*;
+
+#[derive(Debug, Clone, Eq, PartialEq, BorshDeserialize, BorshSerialize, Deserialize, Serialize)]
+pub struct HumanReadableNode {
```
Comment (mariocao) [inline common/src/node/mod.rs:13]: Maybe `NodeInfo`?
```suggestion
pub struct NodeInfo {
```

File: seda_sk
Comment (gluax) [reply seda_sk:1]: whoops! it was ignored in my other branch but not this one 😅

**Review by jamesondh — APPROVED** [review]

**Review by FranklinWaller — APPROVED** [review]

**Pull request merged**